vm/src/stdlib/ctypes/basics.rs

```diff
@@ -0,0 +1,507 @@
+where
+    for<'a> T: PyValue + fmt::Debug + BorrowValue<'a> + BorrowValueMut<'a>,
+{
+    pub data: PyRef<T>,
+    pub options: BufferOptions,
+}
+
+pub struct RawBuffer {
+    pub inner: *mut u8,
+    pub size: usize,
+}
+
+impl fmt::Debug for RawBuffer {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "RawBuffer {{ size: {} }}", self.size)
+    }
+}
+
+impl PyValue for RawBuffer {
+    fn class(vm: &VirtualMachine) -> &PyTypeRef {
+        &vm.ctx.types.object_type
+    }
+}
+
+unsafe impl Send for RawBuffer {}
+unsafe impl Sync for RawBuffer {}
+
+// This Trait is the equivalent of PyCData_Type on tp_base for
+// Struct_Type, Union_Type, PyCPointer_Type
+// PyCArray_Type, PyCSimple_Type, PyCFuncPtr_Type
+#[pyclass(module = "_ctypes", name = "_CData")]
+pub struct PyCData {
+    _objects: AtomicCell<Vec<PyObjectRef>>,
+    _buffer: PyRwLock<RawBuffer>,
+}
+
+impl fmt::Debug for PyCData {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "PyCData {{ _objects: {{}}, _buffer: {{}}}}",)
+    }
+}
+
+impl PyValue for PyCData {
+    fn class(_vm: &VirtualMachine) -> &PyTypeRef {
+        Self::static_type()
+    }
+}
+
+impl PyCData {
+    pub fn new(objs: Option<Vec<PyObjectRef>>, buffer: Option<RawBuffer>) -> Self {
+        PyCData {
+            _objects: AtomicCell::new(objs.unwrap_or_default()),
+            _buffer: PyRwLock::new(buffer.unwrap_or(RawBuffer {
+                inner: ptr::null::<u8>() as *mut _,
+                size: 0,
+            })),
+        }
+    }
+}
+
+#[pyimpl(flags(BASETYPE), with(BufferProtocol))]
+impl PyCData {
+    // PyCData_methods
+    #[pymethod(magic)]
+    pub fn ctypes_from_outparam(zelf: PyRef<Self>) {}
+
+    #[pymethod(magic)]
+    pub fn reduce(zelf: PyRef<Self>) {}
+
+    #[pymethod(magic)]
+    pub fn setstate(zelf: PyRef<Self>) {}
+}
+
+pub fn sizeof_func(tp: Either<PyTypeRef, PyObjectRef>, vm: &VirtualMachine) -> PyResult {
+    match tp {
+        Either::A(type_) if type_.issubclass(PySimpleType::static_type()) => {
+            let zelf = new_simple_type(Either::B(&type_), vm)?;
+            PyCDataFunctions::size_of_instances(zelf.into_ref(vm), vm)
+        }
+        Either::B(obj) if obj.has_class_attr("size_of_instances") => {
+            let size_of = vm.get_attribute(obj, "size_of_instances").unwrap();
+            vm.invoke(&size_of, ())
+        }
+        _ => Err(vm.new_type_error("this type has no size".to_string())),
+    }
+}
+
+pub fn alignment(tp: Either<PyTypeRef, PyObjectRef>, vm: &VirtualMachine) -> PyResult {
+    match tp {
+        Either::A(type_) if type_.issubclass(PySimpleType::static_type()) => {
+            let zelf = new_simple_type(Either::B(&type_), vm)?;
+            PyCDataFunctions::alignment_of_instances(zelf.into_ref(vm), vm)
+        }
+        Either::B(obj) if obj.has_class_attr("alignment_of_instances") => {
+            let size_of = vm.get_attribute(obj, "alignment_of_instances").unwrap();
```

Comment:
```suggestion
            let size_of = vm.get_attribute(obj, "alignment_of_instances")?;
```
is this `unwrap()` intended?